src/build/arg/mod.rs

```diff
@@ -3058,10 +3058,10 @@ impl<'help> Arg<'help> {
     /// # use clap::{App, Arg, ArgSettings};
     /// let res = App::new("prog")
     ///     .arg(Arg::new("cfg")
-    ///         .settings(&[ArgSettings::Required, ArgSettings::TakesValue])
+    ///         .setting(ArgSettings::Required)
     ///         .long("config"))
     ///     .try_get_matches_from(vec![
-    ///         "prog", "--config", "file.conf"
+    ///         "prog", "--config",
```

Comment:
> **Pro tip:** Flags (i.e. not positional, or arguments that take values) shouldn't be
    required by default. This is because if a flag were to be required, it should simply be
    implied. No additional information is required from user. Flags by their very nature are
    simply boolean on/off switches. The only time a user *should* be required to use a flag
    is if the operation is destructive in nature, and the user is essentially proving to you,
    "Yes, I know what I'm doing."

According to comments above, I think the flag in this example would be better to set TakesValue on.